zenoh/src/queryable.rs

```diff
@@ -181,130 +282,66 @@ impl<'a> Resolvable for ReplyBuilder<'a> {
 
 impl SyncResolve for ReplyBuilder<'_> {
     fn res_sync(self) -> <Self as Resolvable>::To {
-        match self.result {
-            Ok(sample) => {
-                if !self.query._accepts_any_replies().unwrap_or(false)
-                    && !self.query.key_expr().intersects(&sample.key_expr)
-                {
-                    bail!("Attempted to reply on `{}`, which does not intersect with query `{}`, despite query only allowing replies on matching key expressions", sample.key_expr, self.query.key_expr())
-                }
-                let Sample {
-                    key_expr,
-                    value: Value { payload, encoding },
-                    kind,
-                    timestamp,
-                    #[cfg(feature = "unstable")]
-                    source_info,
-                    #[cfg(feature = "unstable")]
-                    attachment,
-                } = sample;
-                #[allow(unused_mut)]
-                let mut data_info = DataInfo {
-                    kind,
-                    encoding: Some(encoding),
-                    timestamp,
-                    source_id: None,
-                    source_sn: None,
-                };
-
-                // Use a macro for inferring the proper const extension ID between Put and Del cases
-                macro_rules! ext_attachment {
-                    () => {{
-                        #[allow(unused_mut)]
-                        let mut ext_attachment = None;
+        if !self.query._accepts_any_replies().unwrap_or(false)
+            && !self.query.key_expr().intersects(&self.key_expr)
+        {
+            bail!("Attempted to reply on `{}`, which does not intersect with query `{}`, despite query only allowing replies on matching key expressions", self.key_expr, self.query.key_expr())
+        }
+        #[allow(unused_mut)]
```

Comment:
A comment justifying the `#[allow(unused_mut)]` should be provided.

Reply:
done